### PR TITLE
Improved FE DX

### PR DESCRIPTION
**Pull request by fhenrich33**
Follow up to fix linting errors on CI FE build step: https://github.com/boxwise/boxtribute/pull/1462

Task to implement Turbo repo: https://trello.com/c/r7bet14a/1483-20-fe-migrate-to-turbo-repo

### CODE REVIEW

**Comment by aerinsol** [comment]
@pylipp could you take a neutral look at this and see if the changes here match the ADR (you can mark this as FE hours)? It would be great if you could make a note of any implementation differences to what was proposed in your review as well. If you don't feel like you're the right person for this please feel free to pass it to @HaGuesto . 

@jlin95 , once @pylipp or @HaGuesto have reviewed this, do feel free to do the same an include a functional view by testing it out on your local environment ;)

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1 @@
+npx lint-staged
```
Comment (pylipp) [inline .husky/pre-commit:1]: is there a way to add the `generate-graphql-ts-types` from `front/package.json`?

**Comment by pylipp** [comment]
> @pylipp could you take a neutral look at this and see if the changes here match the ADR (you can mark this as FE hours)? It would be great if you could make a note of any implementation differences to what was proposed in your review as well. If you don't feel like you're the right person for this please feel free to pass it to @HaGuesto .

@aerinsol I can see that husky and lint-staged are in-place as agreed in the ADR. :+1:

File: .husky/pre-commit (hunk 1 above)
Comment (fhenrich33) [reply .husky/pre-commit:1]: I do think so, just need to add the command to the `lint-staged` section in the root `package.json`.

File: package.json [hunk 2]
```diff
@@ -68,14 +71,22 @@
...
-    "@vitest/ui": "1.2.1",
-    "jsdom": "24.0.0"
+    "vitest": "^1.1.3"
+  },
+  "resolutions": {
+    "@graphql-codegen/cli": "^5.0.0",
+    "graphql": "^16.8.1"
+  },
+  "lint-staged": {
+    "*": "pnpm graphql-gen",
```
Comment (pylipp) [inline package.json:88]: I assume the `"*"` indicates the file extension, you can make it `"*.{gql,graphql}"` because it should only react if these filetypes are staged.

File: front/README.md [hunk 3]
```diff
@@ -18,12 +18,6 @@ This front-end project of Boxtribute was bootstrapped with [Create React App](ht
...
 
-### Set-up pre-commit
-
-[Pre-commit](https://pre-commit.com/) enables us to run code quality checks, such as missing semicolons, trailing whitespace, and debug statements, before you are committing your code. We chose pre-commit since it enables us to run these checks for both front-end and back-end in just one place.
-The downside is that you need python to be installed on your computer.
-Please check the [back-end README](../back/README.md#set-up-pre-commit) to set it up.
-
 ### Install node and pnpm
 
 For almost all features of our development set-up you should also have [node](https://nodejs.org/en/download/) installed on your computer. You will need it to run front-end tests and the formatters and linters in your IDE (e.g. VSCode).
```
Comment (HaGuesto) [inline front/README.md:23]: could you add here a line how to install pnpm locally?

File: front/README.md [hunk 4]
```diff
@@ -76,10 +70,14 @@ For example, to add XYZ to the `package.json` file in the `front` folder while d
 Afterwards:
 
 1. stop docker-compose and run `docker compose up` again
-2. run `pnpm` in your local front folder (so that your tooling like VSCode also picks up the changes, like new TS types etc)
+2. run `pnpm i` in your local front folder (so that your tooling like VSCode also picks up the changes, like new TS types etc)
```
Comment (HaGuesto) [inline front/README.md:73]: I do not find this recommendation in BretFisher recommendations. Does running this not mean to install the packages als locally outside the docker container?

File: front/README.md [hunk 5]
```diff
@@ -76,10 +70,14 @@ For example, to add XYZ to the `package.json` file in the `front` folder while d
...
 
 (This advice has come from https://github.com/BretFisher/node-docker-good-defaults)
 
+## About Husky
+
+`husky` is a git hook tool that we use to format and lint staged files on pre-commits. In case you use a version manager tool (e.g. `nvm`, `asdf`), or run into trouble commiting your code, consult https://typicode.github.io/husky/how-to.html#node-version-managers-and-guis.
```
Comment (HaGuesto) [inline front/README.md:79]: to be able to run husky and lint-staged, we need to install node locally, correct?

File: package.json [hunk 6]
```diff
@@ -3,16 +3,18 @@
   "license": "Apache-2.0",
   "packageManager": "pnpm@9.6.0+sha512.38dc6fba8dba35b39340b9700112c2fe1e12f10b17134715a4aa98ccf7bb035e76fd981cf0bb384dfa98f8d6af5481c2bef2f4266a24bfa20c34eb7147ce0b5e",
   "scripts": {
-    "format:check:all": "pnpm format:check \"+(front|shared-components|statviz)/**/*.{js,jsx,json,scss,md,ts,tsx}\" ",
+    "format:check:all": "pnpm format:check \"+(front|shared-components|statviz)/**/*.{js,json,ts,tsx}\" ",
     "format:check": "prettier --check --ignore-path .eslintignore",
-    "format:write:all": "pnpm format:write \"+(front|shared-components|statviz)/**/*.{js,jsx,json,scss,md,ts,tsx}\"",
```
Comment (HaGuesto) [inline package.json:8]: why did you remove jsx here?

File: front/README.md (hunk 4 above)
Comment (fhenrich33) [reply front/README.md:73]: You need to, just like you need Python locally to have your tools available. If you are developing for the frontend, having Node.js et al is a given.

File: front/README.md (hunk 5 above)
Comment (fhenrich33) [reply front/README.md:79]: That's correct @HaGuesto .

File: package.json (hunk 6 above)
Comment (fhenrich33) [reply package.json:8]: We don't use JSX ever, and if a package uses it, we don't format it ourselves.

File: package.json [hunk 7]
```diff
@@ -68,14 +71,22 @@
...
     "typescript": "^5.3.3",
     "vite": "^5.3.4",
     "vite-tsconfig-paths": "^4.2.3",
-    "vitest": "^1.1.3",
-    "@vitest/coverage-v8": "1.2.1",
-    "@vitest/ui": "1.2.1",
-    "jsdom": "24.0.0"
+    "vitest": "^1.1.3"
+  },
+  "resolutions": {
```
[the remark this answers is not in the document]
Comment (fhenrich33) [reply package.json:83]: This or `overrides` are meant to resolve interdependencies conflicts, using this field as the intended version over what any given package wants to install as their deps. https://pnpm.io/package_json#resolutions

File: package.json (hunk 2 above)
Comment (fhenrich33) [reply package.json:88]: @pylipp @HaGuesto  Quick note on that: you might need to have the local frontend toolchain if you want to generate these files on commit.
